transform-plugins/src/main/java/io/cdap/plugin/Hasher.java

```diff
@@ -124,14 +131,37 @@ public Config(String hash, String fields) {
       this.fields = fields;
     }
 
-    private void validate() {
+    private void validate(@Nullable Schema inputSchema, FailureCollector failureCollector) {
       // Checks if hash specified is one of the supported types.
-      if (!hash.equalsIgnoreCase("md2") && !hash.equalsIgnoreCase("md5") &&
+      if (hash != null && !hash.equalsIgnoreCase("md2") && !hash.equalsIgnoreCase("md5") &&
         !hash.equalsIgnoreCase("sha1") && !hash.equalsIgnoreCase("sha256") &&
         !hash.equalsIgnoreCase("sha384") && !hash.equalsIgnoreCase("sha512")) {
-        throw new IllegalArgumentException("Invalid hasher '" + hash + "' specified. Allowed hashers are md2, " +
-                                             "md5, sha1, sha256, sha384 and sha512");
+        failureCollector.addFailure(String.format("Invalid hasher '%s' specified.", hash),
+                                    "Allowed hashers are md2, md5, sha1, sha256, sha384 and sha512");
       }
+
+      if (inputSchema == null) {
+        return;
+      }
+
+      for (String field : getFields()) {
+        Schema.Field inputField = inputSchema.getField(field);
+        if (inputField == null) {
```

Comment:
Isn't this error case where field to be hashed is not present in input?

Reply:
The current behavior is that it gets ignored, so I didn't want to change it.